tst/com/amazon/corretto/crypto/provider/test/EvpSignatureSpecificTest.java

```diff
@@ -638,6 +638,36 @@ public void ecdsaSignCorruptsErrorState() throws Exception {
     }
   }
 
+  /**
+   * For better compatibility with the JDK, we want to throw SignatureException if the length of the
+   * signature is wrong.
+   */
+  @Test
+  public void rsaSignatureWrongLength() throws Exception {
+    final Signature signer = Signature.getInstance("SHA256withRSA", NATIVE_PROVIDER);
+    signer.initSign(RSA_PAIR.getPrivate());
+    signer.update(MESSAGE);
+    final byte[] validSignature = signer.sign();
+
+    // Sanity check
+    signer.initVerify(RSA_PAIR.getPublic());
+    signer.update(MESSAGE);
+    assertTrue(signer.verify(validSignature));
+
+    // Too long throws an exception
+    final byte[] longSig = Arrays.copyOf(validSignature, validSignature.length + 1);
+    longSig[longSig.length - 1] = 1;
```

Comment:
why even fill this in?

Reply:
To ensure that we don't just consume the valid prefix and incorrectly accept.